EasyPost/Utilities/Internal/Extensions/Enumerables.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+
+namespace EasyPost.Utilities.Internal.Extensions
+{
+    public static class Enumerables
+    {
+        /// <summary>Foreach with an index.</summary>
+        /// <param name="ie">IEnumerable element.</param>
+        /// <param name="action">
+        ///     Action to do for each loop.
+        ///     Action receives a (int, T) tuple containing the index and the current element in the loop as a parameter
+        /// </param>
+        /// <typeparam name="T">Type of input.</typeparam>
+        public static void Each<T>(this IEnumerable<T> ie, Action<int, T> action)
```

Comment:
### Author's Note

This extension method is used to ease the process of conducting an action on each element in a list. Specifically, this is used by the `toDictionary` method override for making a tracker list.